rest/src/test/scala/io/udash/rest/openapi/RestSchemaTest.scala

```diff
@@ -71,6 +71,22 @@ object FullyQualifiedHierarchy extends RestDataCompanionWithDeps[FullyQualifiedN
   case object Baz extends FullyQualifiedHierarchy
 }
 
+@flatten("case")
+sealed trait CustomSchemaNameHierarchy
+object CustomSchemaNameHierarchy extends RestDataCompanion[CustomSchemaNameHierarchy] {
+  // annotation value should be used as schema name, but NOT as type discriminator value
+  @schemaName("CustomSchemaName123")
+  case class CustomSchemaName(str: String) extends CustomSchemaNameHierarchy
+
+  // annotation value should be used as both schema name and type discriminator value
+  @name("CustomName123")
+  case class CustomName(str: String) extends CustomSchemaNameHierarchy
+
+  // @schemaName should "win" with @name as schema name
```

Comment:
`@name as schema name` or `@name as type dyscriminator value`? When I look at the declaration I'd expect the latter :D